apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/codegen/CodegenLayout.kt

```diff
@@ -53,6 +46,18 @@ internal abstract class CodegenLayout(
       usedNames.add(className.lowercase())
       this[type.name] = className
     }
+
+    // 2. Use targetName verbatim for types that define it
+    // Note: prior validation in checkApolloDuplicateTargetNames checks for clashes but there may still
+    // be a clash due to step 1.
```

Comment:
Consolidate both validations in a single method? Not only is it less code but it'll allow displaying a nice `SourceAwareException` with some context if we ever end up in this case?